frontend/ui/widget/keyvaluepage.lua

```diff
@@ -495,11 +501,21 @@ function KeyValuePage:init()
     local available_height = self.inner_dimen.h
                          - self.title_bar:getSize().h
                          - Size.span.vertical_large -- for above page_info (as title_bar adds one itself)
-                         - self.page_info:getSize().h
+                         - (self.single_page and 0 or self.page_info:getSize().h)
                          - 2*Size.line.thick
                             -- account for possibly 2 separator lines added
 
-    self.items_per_page = G_reader_settings:readSetting("keyvalues_per_page") or self:getDefaultKeyValuesPerPage()
+    local force_items_per_page
+    if self.single_page then
+        if not self.min_nb_items then
+            force_items_per_page = #self.kv_pairs
+        else
+            force_items_per_page = math.max(#self.kv_pairs, self.min_nb_items)
```

Comment:
I'm not sure `min_nb_items` is a good idea: when would we developpers feel like we have to use it, and with which hardcoded value ? and if not hardcoded, a value based on which other thingy ?

I would probably just use:
```lua
force_items_per_page = math.max(#self.kv_pairs,
    G_reader_settings:readSetting("keyvalues_per_page") or self:getDefaultKeyValuesPerPage() )
```
That is: we know the user is fine reading with its chosen `G_reader_settings:readSetting("keyvalues_per_page")`, so it's ok using it if we have less items - he will get some blank area at bottom of the page, no real need to zoom everything.
(But with `self.single_page` we just hope he will be fine if we go a little smaller to fit everything on one page.)